test/widgets/board_test.dart

```diff
@@ -106,6 +106,21 @@ void main() {
       expect(find.byKey(const Key('e4-lastMove')), findsOneWidget);
     });
 
+    testWidgets('Cannot move by tap if piece shift method is drag',
+        (WidgetTester tester) async {
+      await tester.pumpWidget(
+        buildBoard(
+          initialInteractableSide: InteractableSide.both,
+          pieceShiftMethod: PieceShiftMethod.drag,
+        ),
+      );
+      await tester.tap(find.byKey(const Key('e2-whitePawn')));
+      await tester.pump();
+
+      expect(find.byKey(const Key('e2-selected')), findsNothing);
+      expect(find.byType(MoveDest), findsNothing);
```

Comment:
I think we should add a test below that verify move by drag works when `pieceShiftMethod` is drag.